qutebrowser/browser/history.py

```diff
@@ -288,14 +185,74 @@ def add_url(self, url, title="", *, redirect=False, atime=None):
         if atime is None:
             atime = time.time()
         entry = Entry(atime, url, title, redirect=redirect)
-        if self._initial_read_done:
-            self._add_entry(entry)
-            self._new_history.append(entry)
-            self.item_added.emit(entry)
-            if not entry.redirect:
-                self.add_completion_item.emit(entry)
+        self._add_entry(entry)
+
+    def _parse_entry(self, line):
+        """Parse a history line like '12345 http://example.com title'."""
+        data = line.split(maxsplit=2)
+        if len(data) == 2:
+            atime, url = data
+            title = ""
+        elif len(data) == 3:
+            atime, url, title = data
+        else:
+            raise ValueError("2 or 3 fields expected")
+
+        url = QUrl(url)
+        if not url.isValid():
```

Comment:
We might want to special case `http://.com/` and `https://www..com/` here and silently ignore them - they could be added to the history because of [this QUrl bug](https://bugreports.qt.io/browse/QTBUG-60364), and were used in the fake Apple phishing domain http://xn--pple-43d.com/ which was in the media, so it's entirely possible this is a common annoyance. But since those history entries don't carry any value, it should be safe to skip them.